Sources/DataPipeline/CustomerIO+Events.swift

```
@@ -37,6 +37,6 @@ public extension CustomerIO {
     ///   - category: A category to the type of screen if it applies.
     ///   - properties: Any extra metadata associated with the screen. e.g. method of access, size, etc.
     func screen(title: String, category: String? = nil, properties: [String: Any]? = nil) {
-        DataPipeline.shared.analytics.screen(title: title, category: category, properties: properties)
+        DataPipeline.shared.analytics.screen(title: title, category: category, properties: properties?.sanitizedForJSON())
```

I don't think this is the right place to add protection.
It should be added inside Analytics implementation. That way every call to analytics is protected.
In this same file, you have call to analytics (method `func screen<P: Codable>(title: String, category: String? = nil, properties: P?)`) that isn't protected.

Let's make sure that all methods which are working with `properties`/`traits` in Analytics are protected. This applies to both of those that are using Dictionary and Generics (where `JSON(with:)` is called - shouldn't silently fail).